src/Microsoft.Extensions.DependencyInjection/ServiceLookup/ClosedIEnumerableService.cs

```diff
@@ -3,19 +3,18 @@
 
 using System;
 using System.Collections.Generic;
-using System.Linq;
 
 namespace Microsoft.Extensions.DependencyInjection.ServiceLookup
 {
     internal class ClosedIEnumerableService : IService
     {
-        private readonly Type _itemType;
         private readonly ServiceEntry _serviceEntry;
 
         public ClosedIEnumerableService(Type itemType, ServiceEntry entry)
         {
-            _itemType = itemType;
+            ServiceType = itemType;
             _serviceEntry = entry;
+            ImplementationType = itemType.MakeArrayType();
```

Comment:
So we don't generate `provider.CaptureDisposable(new T[] { ... })` when generating the code for `IEnumerable<T>`. See the PR description for an example (it's the first one).